libs/estdlib/src/erlang.erl

```diff
@@ -761,6 +762,17 @@ float_to_list(_Float) ->
 float_to_list(_Float, _Options) ->
     erlang:nif_error(undefined).
 
+%%-----------------------------------------------------------------------------
+%% @param   Fun     Function to get information about
+%% @param   Info    A list of atoms specifying the information to return.
+%%                  Available atoms are: module, name, arity, type
+%% @returns         Requested information about the function as a list of tuples.
+%% @doc             Returns specific information about the function `Fun` in unspecified order.
+%% @end
+%%-----------------------------------------------------------------------------
+fun_info(_Fun, _Info) ->
+    erlang:nif_error(undefined).
```

Comment:
Where are those definitions replaced? In `module_build_imported_functions_table`?

I wanted to implement `fun_info/1` in terms of `fun_info/2` directly in Erlang but it didn't work.

Reply:
You can do this, but you need to write the call `?MODULE:fun_info/2` so it performs the lookup.
Nowadays, I rather implement all arities in C and use argc.

Reply:
Thanks! For this function it made more sense that way since it's not typically used outside of code doing reflection for whatever reason which should be rare (and it's easier in Erlang due to the GC thing mentioned below).